source/backend/database/models/recipesModel.js

```diff
@@ -55,6 +55,24 @@ async function createRecipe(payload, ingredientsArray) {
 async function deleteRecipe(userId, recipeId) {
   await db.transaction(async (transaction) => {
     try {
+      const ingredientsArray = 
+        await db('recipeIngredients')
+            .select('*')
+            .where({recipeId});
+      console.log(ingredientsArray[0].recipeId);
+      await Promise.all(ingredientsArray.map(async (ingredient) => {
```

Comment:
Instead of iterating through an ingredientsArray, and call del on database, we can reduce the overhead to O(1) for database calls. 
1. For the `recipeIngredients` table, a simple delete by recipeId should work.
2.  mysql supports a delete with a `whereIn` function that allows us to pass in an array of elements. We can use that to batch delete ingredients.

```
await db('recipeIngredients')
	.transacting(transaction)
	.where({recieId})
	.del();

// you will have an array filled with ingredient ids: eg [1, 2, 3...]
const ingredientIds = await Promise.all(ingredientsArray.map(recipeIngredient => recipeIngredient.ingredientId));

await db('ingredients')
	.transacting(transaction)
	.whereIn('id', ingredientIds)
	.del();
```

Reply:
Fixed! You're a genius!